cats_frontend/src/app/modules/cats/pages/cats-list/cats-list.component.ts

```diff
@@ -1,55 +1,57 @@
-import { Component, OnInit } from "@angular/core";
+import { Component, OnDestroy, OnInit } from "@angular/core";
 import { ICat } from "../../models/cat.model";
-import { CatsService } from "../../services/cats.service";
 import { ActivatedRoute, Router } from "@angular/router";
-import { CommentsService } from "../../services/comments.service";
 import { IComment } from "../../models/comment.model";
 import { FormGroup, FormControl, Validators } from "@angular/forms";
-import { FilterMatchMode, MessageService, SelectItem } from "primeng/api";
+import { FilterMatchMode } from "primeng/api";
 import { DialogService } from 'primeng/dynamicdialog';
 import { CreateCommentComponent } from "../../components/modals/create-comment/create-comment.component";
 import { EditCommentComponent } from "../../components/modals/edit-comment/edit-comment.component";
-import { Subscription, debounceTime, distinctUntilChanged } from "rxjs";
+import { Observable, Subject, debounceTime, distinctUntilChanged, takeUntil } from "rxjs";
 import { DatePipe } from "@angular/common";
+import { Store, select } from "@ngrx/store";
+import { CatsActions } from "../../store/actions/cats.actions";
+import { CommentsActions } from "../../store/actions/comments.actions";
+import { selectCatWithComments, selectCount } from "../../store/selectors/cats.selectors";
+import { cloneDeep } from 'lodash';
 
 @Component({
     selector: 'app-cats-list',
     templateUrl: './cats-list.component.html'
 })
-export class CatsListComponent implements OnInit {
+export class CatsListComponent implements OnInit, OnDestroy {
+
+     // subject
+     private _unsubscribe$$ = new Subject<void>();
+
+    // observables
+    cats$: Observable<ICat[]> = this.store.pipe(takeUntil(this._unsubscribe$$), select(selectCatWithComments));
+    count$: Observable<number> = this.store.pipe(takeUntil(this._unsubscribe$$), select(selectCount));
 
     // booleans
     editSidebarIsOpened: boolean = false;
-    createCommentModalIsOpened: boolean = false;
-    editCommentModalIsOpened: boolean = false;
-
-    // expanded
-    selectedCat: ICat | null = null;
     
     // table options
-    cats: ICat[] = [];
```

Comment:
change this to an observable that's connected to the right selectors in your store